### PR TITLE
[zh-cn]: init Barcode Detection API

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

see https://developer.mozilla.org/en-US/docs/Web/API/Barcode_Detection_API

### CODE REVIEW

File: files/zh-cn/web/api/barcode_detection_api/index.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+{{securecontext_header}}{{DefaultAPISidebar("Barcode Detection API")}}{{AvailableInWorkers}}{{SeeCompatTable}}
+
+Barcode Detection API 可用于检测图像中的条形码和二维码。
+
+## 概念和用法
+
+Web 应用程序中对条形码识别的支持通过支持的各种条形码格式解锁了各种用例。二维码可用于在线支付、网络导航或建立社交媒体连接，Aztec 码可用于扫描登机牌，购物应用程序可使用 EAN 或 UPC 条形码来比较实物商品的价格。
+
+检测是通过 {{domxref('BarcodeDetector.detect()','detect()')}} 方法实现的，该方法接受一个图像对象；它可以是以下对象之一：一个 {{domxref("HTMLImageElement")}}、一个 {{domxref("SVGImageElement")}}、一个 {{domxref("HTMLVideoElement")}}、一个 {{domxref("HTMLCanvasElement")}}、一个 {{domxref("ImageBitmap")}}、一个 {{domxref("OffscreenCanvas")}}、一个 {{domxref("VideoFrame")}}、一个 {{domxref('Blob')}}、或一个 {{domxref('ImageData')}}。可以将可选参数传递给 {{domxref('BarcodeDetector')}} 构造函数，以提供有关要检测的条形码格式的提示。
```
Comment (yin1999) [inline files/zh-cn/web/api/barcode_detection_api/index.md:16]: remove quantifiers

```suggestion
检测是通过 {{domxref('BarcodeDetector.detect()','detect()')}} 方法实现的，该方法接受一个图像对象；它可以是以下对象之一：{{domxref("HTMLImageElement")}}、{{domxref("SVGImageElement")}}、{{domxref("HTMLVideoElement")}}、{{domxref("HTMLCanvasElement")}}、{{domxref("ImageBitmap")}}、{{domxref("OffscreenCanvas")}}、{{domxref("VideoFrame")}}、{{domxref('Blob')}} 或 {{domxref('ImageData')}}。可以将可选参数传递给 {{domxref('BarcodeDetector')}} 构造函数，以提供有关要检测的条形码格式的提示。
```

File: files/zh-cn/web/api/barcode_detection_api/index.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        <img
+          alt="一个 code 93 条形码的图像。一个垂直黑白线的水平布局图案"
+          src="code-93.png"
+        />
+      </td>
+    </tr>
+    <tr>
+      <td>codabar</td>
+      <td>
+        一种表示字符 0-9、A-D 和符号 - $ / + 的线性条形码。
```
Comment (yin1999) [inline files/zh-cn/web/api/barcode_detection_api/index.md:82]:
```suggestion
        一种表示字符 0-9、A-D 和符号 - . $ / + 的线性条形码。
```

File: files/zh-cn/web/api/barcode_detection_api/index.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+        <img
+          alt="一个 upc-e 条形码的图像。一个由黑白垂直线组成的矩形"
+          src="upc-e.png"
+        />
+      </td>
+    </tr>
+    <tr>
+      <td>unknown</td>
+      <td>
+        平台使用此值来表示它不知道或指定目前正在检测或支持哪种条形码格式。
```
Comment (yin1999) [inline files/zh-cn/web/api/barcode_detection_api/index.md:188]:
```suggestion
        平台使用此值来表示它不知道或未指定目前正在检测或支持的条形码格式。
```

File: files/zh-cn/web/api/barcode_detection_api/index.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+      </td>
+      <td></td>
+    </tr>
+  </tbody>
+</table>
+
+你可以通过 {{domxref('BarcodeDetector/getSupportedFormats_static','getSupportedFormats()')}} 方法检查用户代理支持识别的格式。
```
Comment (yin1999) [inline files/zh-cn/web/api/barcode_detection_api/index.md:195]:
```suggestion
你可以通过 {{domxref('BarcodeDetector/getSupportedFormats_static','getSupportedFormats()')}} 方法检查用户代理支持的格式。
```

File: files/zh-cn/web/api/barcode_detection_api/index.md [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+## 接口
+
+- {{domxref("BarcodeDetector")}} {{Experimental_Inline}}
+  - : Barcode Detection API 的 **`BarcodeDetector`** 接口允许检测图像中的条形码和二维码。
```
Comment (yin1999) [inline files/zh-cn/web/api/barcode_detection_api/index.md:200]: We could remove the bold style, cloud you fix the English documentation?

```suggestion
  - : Barcode Detection API 的 `BarcodeDetector` 接口允许检测图像中的条形码和二维码。
```

**Review by yin1999 — COMMENTED** [review]
Thanks @skyclouds2001. Just some improvement:

File: files/zh-cn/web/api/barcode_detection_api/index.md (hunk 5 above)
Comment (skyclouds2001) [reply files/zh-cn/web/api/barcode_detection_api/index.md:200]: see https://github.com/mdn/content/pull/30976

File: files/zh-cn/web/api/barcode_detection_api/index.md [hunk 6]
```diff
@@ -0,0 +1,259 @@
+---
+title: Barcode Detection API
+slug: Web/API/Barcode_Detection_API
+l10n:
+  sourceCommit: 4458494807b6f4898d504b6c0af0a45f8031cbf3
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/barcode_detection_api/index.md:5]:
```suggestion
l10n:
  sourceCommit: 41d343d684f9f6e7199d408b209bcd0e077eb023
```

update for https://github.com/mdn/content/pull/30976

**Review by yin1999 — APPROVED** [review]
Thank you @skyclouds2001 :+1:

**Pull request merged**